src/main/java/org/opensearch/security/filter/SecurityResponse.java

```diff
@@ -54,9 +55,22 @@ public String getBody() {
     }
 
     public RestResponse asRestResponse() {
-        final RestResponse restResponse = new BytesRestResponse(RestStatus.fromCode(getStatus()), getBody());
+        String contentType = null;
         if (getHeaders() != null) {
-            getHeaders().forEach(restResponse::addHeader);
+            contentType = getHeaders().get(HttpHeaders.CONTENT_TYPE);
+        }
+        RestResponse restResponse;
+        if (contentType != null) {
+            restResponse = new BytesRestResponse(RestStatus.fromCode(getStatus()), contentType, getBody());
+        } else {
+            restResponse = new BytesRestResponse(RestStatus.fromCode(getStatus()), getBody());
+        }
+        if (getHeaders() != null) {
+            getHeaders().entrySet()
+                .stream()
+                .filter(e -> !e.getKey().equalsIgnoreCase(HttpHeaders.CONTENT_TYPE))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+                .forEach(restResponse::addHeader);
```

Comment:
Thanks for the quick work! However, this will only be a quick fix to support content-type header. However, this needs to be more generic, you can look at all the different constructors core OS class supports- https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/rest/BytesRestResponse.java

Before the recent refactoring, multiple different BytesRestResponse calls were part of security plugin code. Also, we have not replaced all the APIs, e.g. https://github.com/opensearch-project/security/blob/2.11/src/main/java/org/opensearch/security/rest/SecurityInfoAction.java#L96

If we look to port these APIs as well, you will see the need to have more generic constructors like BytesRestResponse

We need to write a better wrapper here.

Reply:
Got it, I just pushed a change to explicitly pass contentType in the SecurityResponse constructor. Similar to the BytesResponseResponse constructor.